src/memkind_dax_kmem.c

```diff
@@ -172,13 +180,21 @@ static void memkind_dax_kmem_closest_numanode_init(void)
     struct dax_closest_numanode_t *g = &memkind_dax_kmem_closest_numanode_g;
     int *bandwidth = (int *)jemk_calloc(NUMA_NUM_NODES, sizeof(int));
     int num_unique = 0;
+    int i;
 
     struct bandwidth_nodes_t *bandwidth_nodes = NULL;
 
+    int num_nodes = numa_num_configured_nodes();
+
     g->num_cpu = numa_num_configured_cpus();
-    g->closest_numanode = (int *)jemk_malloc(sizeof(int) * g->num_cpu);
+    g->closest_numanodes = (int **)jemk_malloc(sizeof(int *) * g->num_cpu +
```

Comment:
At this point, you might as well make this more readable and fold allocation/initialization of the matrix into the function itself.